File: 2022/221222-dojo/javascript/solution.js
export default function solution(array) {
  return array.map((number, index) => {
    if (index === 0) {
      return number;
    }

    if (index !== 0 && array[index - 1] !== number) {
      return number;
    }

    return '';
  })
    .filter((value) => value !== '');
}

// export default function solution(array) {
//     return array.filter((value, index) => value !== array[index + 1]);
//   }

File: 2022/221222-dojo/java/app/src/main/java/Solution.java
import java.util.*;

public class Solution {
    public int[] solution(int []arr) {
        Stack<Integer> stack = new Stack<>();

        for (int number : arr) {
            if (stack.empty()) {
                stack.push(number);
                continue;
            }

            if (!stack.peek().equals(number)) {
                stack.push(number);
            }
        }

        int[] newArray = new int[stack.size()];

        for (int i = newArray.length - 1; i >= 0; i -= 1) {
            newArray[i] = stack.pop();
        }

        return newArray;
    }
}
